packages/deployment-service/src/marketplace/marketplace.provider.ts

```diff
@@ -41,44 +41,48 @@ export class MarketplaceProvider {
     return { access_token, token_type }
   }
 
-  async getAppDetails(appId: string): Promise<Marketplace.AppDetailModel> {
-    const { access_token, token_type } = await this.authenticate()
-
+  /**
+   * Find the app's latest revision for patch update
+   *
+   * @param appId
+   * @param token_type
+   * @param access_token
+   * @returns Revision or undefined
+   */
+  private async findLatestRevision(
+    appId: string,
+    token_type: string,
+    access_token: string,
+  ): Promise<Marketplace.AppRevisionModel | undefined> {
     const result = await firstValueFrom(
-      this.httpService.get<Marketplace.AppDetailModel>(`${this.config.url}/apps/${appId}`, {
+      this.httpService.get<Marketplace.AppRevisionModelPagedResult>(`${this.config.url}/apps/${appId}/revisions`, {
         headers: {
           Authorization: `${token_type} ${access_token}`,
           ['api-version']: 1,
         },
       }),
     )
 
-    return result.data
+    return result?.data?.data ? result.data.data[0] : undefined
```

Comment:
This might need to be a bit more intelligent. If there's two revisions you need to make sure you're choosing the latest one (which will be in a pending status). This won't necessarily appear first in the collection

Reply:
Just checked the code and these results are sorted in date order, so if there's a pending one it'll always appear at the top so you're fine with this